File: metacasa-ios/MetaCasa/Models/TransactionTemplate.swift
import Foundation

/// Plantilla de transacción guardada por el usuario para crear movimientos
/// recurrentes con 1 tap. Port del feature "quick shortcuts" del web
/// (App.jsx:3229-3244).
struct TransactionTemplate: Codable, Identifiable, Hashable, Sendable {
    let id: UUID
    var householdId: UUID
    var name: String
    var emoji: String?
    var type: TxType
    var amount: Decimal
    var currency: String
    var category: String
    var subcategory: String?
    var note: String?
    var position: Int
    let createdBy: UUID
    let createdAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case householdId = "household_id"
        case name
        case emoji
        case type
        case amount
        case currency
        case category
        case subcategory
        case note
        case position
        case createdBy = "created_by"
        case createdAt = "created_at"
    }
}

File: metacasa-flutter/lib/models/transaction_template.dart
import 'package:decimal/decimal.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

import '../core/utils/decimal_converter.dart';
import 'transaction.dart';

part 'transaction_template.freezed.dart';
part 'transaction_template.g.dart';

/// Plantilla de transacción guardada para crear movimientos con 1 tap
/// ("quick shortcuts"). Espejo 1:1 de `public.transaction_templates`.
@freezed
abstract class TransactionTemplate with _$TransactionTemplate {
  const factory TransactionTemplate({
    required String id,
    @JsonKey(name: 'household_id') required String householdId,
    required String name,
    String? emoji,
    required TxType type,
    @DecimalConverter() required Decimal amount,
    required String currency,
    required String category,
    String? subcategory,
    String? note,
    required int position,
    @JsonKey(name: 'created_by') required String createdBy,
    @JsonKey(name: 'created_at') DateTime? createdAt,
  }) = _TransactionTemplate;

  factory TransactionTemplate.fromJson(Map<String, dynamic> json) =>
      _$TransactionTemplateFromJson(json);
}
